src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.State.cs

```diff
@@ -110,6 +110,8 @@ internal sealed class State
             /// The <see cref="DateTime"/> values here do not update dynamically.
             /// </remarks>
             public IImmutableDictionary<string, DateTime> AdditionalDependentFileTimes { get; }
+            public ImmutableHashSet<string> LastAdditionalDependentFileTimes { get; set; }
+            public bool AdditionalDependentFilesAddedOrRemoved;
```

Comment:
The state object must be immutable. That means the property cannot have a setter, and the field must be a get-only property.

Reply:
These two fields are redundant as the bool can be computed in the state update method without requiring memory of the last set.

I think it would be better to remove these new properties and store `LastAdditionalDependentFileTimesChangedAtUtc` instead, matching the pattern used in `LastItemsChangedAtUtc`.